### ISSUE TITLE
Error parsing flag: -g

### ISSUE THREAD

**Issue by GuzioMG** [espidev/ProtectionStones#281]
Hello. I recently (finally) configured default settings for ProtectionStones WG flags on my server, but it appears as though it doesn't work.

A little bit of context: I assigned a lot of flags to groups, so I used the  `-g` param quite a lot. According to the docs, it's perfectly fine:

> Can use -g [group] before the flag to set group flags (ex. -g members pvp deny).

and

> If you want to whitelist the groups that can be set as well, use -g (ex. "-g all,members pvp" restricts it to no group flag, and members group)

The problem is that I get the following error on the server startup:

_(only attaching the part regarding PStones)_

```
30.07 20:45:49 [Server] INFO [ClearLag] No updates found!
30.07 20:45:50 [Server] INFO [ProtectionStones] Enabling ProtectionStones v2.8.5
30.07 20:45:50 [Server] INFO [WorldGuard] Registering session handler dev.espi.protectionstones.flags.GreetingFlagHandler
30.07 20:45:50 [Server] INFO [WorldGuard] Registering session handler dev.espi.protectionstones.flags.FarewellFlagHandler
30.07 20:45:50 [Server] WARN [ProtectionStones] No economy plugin found by Vault! There will be no economy support!
30.07 20:45:50 [Server] INFO [ProtectionStones] PlaceholderAPI not found! There will be no PlaceholderAPI support.
30.07 20:45:50 [Server] INFO [ProtectionStones] LuckPerms support enabled!
30.07 20:45:51 [Server] INFO [ProtectionStones] Protection Stone Blocks:
30.07 20:45:51 [Server] INFO [ProtectionStones] - RAW_GOLD_BLOCK (mk-I)
30.07 20:45:51 [Server] WARN [ProtectionStones] Error parsing flag: -g
30.07 20:45:51 [Server] INFO Error: 
30.07 20:45:51 [Server] WARN java.lang.NullPointerException: Cannot invoke "com.sk89q.worldguard.protection.flags.Flag.parseInput(com.sk89q.worldguard.protection.flags.FlagContext)" because "flag" is null
30.07 20:45:51 [Server] WARN at dev.espi.protectionstones.FlagHandler.initDefaultFlagsForBlock(FlagHandler.java:208)
30.07 20:45:51 [Server] WARN at dev.espi.protectionstones.PSConfig.initConfig(PSConfig.java:211)
30.07 20:45:51 [Server] WARN at dev.espi.protectionstones.ProtectionStones.loadConfig(ProtectionStones.java:492)
30.07 20:45:51 [Server] WARN at dev.espi.protectionstones.ProtectionStones.onEnable(ProtectionStones.java:591)
30.07 20:45:51 [Server] WARN at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
30.07 20:45:51 [Server] WARN at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370)
30.07 20:45:51 [Server] WARN at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500)
30.07 20:45:51 [Server] WARN at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:518)
30.07 20:45:51 [Server] WARN at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:432)
30.07 20:45:51 [Server] WARN at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:639)
30.07 20:45:51 [Server] WARN at net.minecraft.server.dedicated.DedicatedServer.init(DedicatedServer.java:306)
30.07 20:45:51 [Server] WARN at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1126)
30.07 20:45:51 [Server] WARN at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316)
30.07 20:45:51 [Server] WARN at java.base/java.lang.Thread.run(Thread.java:831)
30.07 20:45:51 [Server] INFO [ProtectionStones] - BUDDING_AMETHYST (mk-II)
30.07 20:45:51 [Server] WARN [ProtectionStones] Error parsing flag: -g
30.07 20:45:51 [Server] INFO Error: 
30.07 20:45:51 [Server] WARN java.lang.NullPointerException: Cannot invoke "com.sk89q.worldguard.protection.flags.Flag.parseInput(com.sk89q.worldguard.protection.flags.FlagContext)" because "flag" is null
30.07 20:45:51 [Server] WARN at dev.espi.protectionstones.FlagHandler.initDefaultFlagsForBlock(FlagHandler.java:208)
30.07 20:45:51 [Server] WARN at dev.espi.protectionstones.PSConfig.initConfig(PSConfig.java:211)
30.07 20:45:51 [Server] WARN at dev.espi.protectionstones.ProtectionStones.loadConfig(ProtectionStones.java:492)
30.07 20:45:51 [Server] WARN at dev.espi.protectionstones.ProtectionStones.onEnable(ProtectionStones.java:591)
30.07 20:45:51 [Server] WARN at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
30.07 20:45:51 [Server] WARN at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370)
30.07 20:45:51 [Server] WARN at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500)
30.07 20:45:51 [Server] WARN at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:518)
30.07 20:45:51 [Server] WARN at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:432)
30.07 20:45:51 [Server] WARN at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:639)
30.07 20:45:51 [Server] WARN at net.minecraft.server.dedicated.DedicatedServer.init(DedicatedServer.java:306)
30.07 20:45:51 [Server] WARN at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1126)
30.07 20:45:51 [Server] WARN at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316)
30.07 20:45:51 [Server] WARN at java.base/java.lang.Thread.run(Thread.java:831)
30.07 20:45:51 [Server] INFO [ProtectionStones] - LODESTONE (mk-III)
30.07 20:45:51 [Server] WARN [ProtectionStones] Error parsing flag: -g
30.07 20:45:51 [Server] INFO Error: 
30.07 20:45:51 [Server] WARN java.lang.NullPointerException: Cannot invoke "com.sk89q.worldguard.protection.flags.Flag.parseInput(com.sk89q.worldguard.protection.flags.FlagContext)" because "flag" is null
30.07 20:45:51 [Server] WARN at dev.espi.protectionstones.FlagHandler.initDefaultFlagsForBlock(FlagHandler.java:208)
30.07 20:45:51 [Server] WARN at dev.espi.protectionstones.PSConfig.initConfig(PSConfig.java:211)
30.07 20:45:51 [Server] WARN at dev.espi.protectionstones.ProtectionStones.loadConfig(ProtectionStones.java:492)
30.07 20:45:51 [Server] WARN at dev.espi.protectionstones.ProtectionStones.onEnable(ProtectionStones.java:591)
30.07 20:45:51 [Server] WARN at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
30.07 20:45:51 [Server] WARN at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370)
30.07 20:45:51 [Server] WARN at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:500)
30.07 20:45:51 [Server] WARN at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugin(CraftServer.java:518)
30.07 20:45:51 [Server] WARN at org.bukkit.craftbukkit.v1_17_R1.CraftServer.enablePlugins(CraftServer.java:432)
30.07 20:45:51 [Server] WARN at net.minecraft.server.MinecraftServer.loadWorld(MinecraftServer.java:639)
30.07 20:45:51 [Server] WARN at net.minecraft.server.dedicated.DedicatedServer.init(DedicatedServer.java:306)
30.07 20:45:51 [Server] WARN at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1126)
30.07 20:45:51 [Server] WARN at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316)
30.07 20:45:51 [Server] WARN at java.base/java.lang.Thread.run(Thread.java:831)
30.07 20:45:51 [Server] WARN [ProtectionStones] Vault is not enabled! Economy functions (renting & buying) will not work!
30.07 20:45:51 [Server] INFO [ProtectionStones] Building region cache...
30.07 20:45:51 [Server] INFO [ProtectionStones] Building UUID cache... (if slow change async-load-uuid-cache in the config to true)
30.07 20:45:51 [Server] INFO [ProtectionStones] Checking if PS regions have been updated to UUIDs...
30.07 20:45:51 [Server] INFO [ProtectionStones] ProtectionStones has successfully started!
30.07 20:45:52 [Server] INFO Running delayed init tasks
30.07 20:45:52 [Server] Startup Done (35.698s)! For help, type "help"
30.07 20:45:52 [Server] INFO Timings Reset
```

This error doesn't seem to be game-breaking, as all the stones load correctly and only flags that should be editable can be edited and only for groups they are supposed to, BUT all the defaults are gone and set to unreasonable values (eg. members can't build). Currently, I mitigated the issue by sending a message in chat to players when they create a region to manually adjust flags, but that's rather inconvenient. Bellow, I'm attaching my config regarding flags. It's the exact same for all the blocks.

```
# Specify the default flags to be set when a new protected region is created.
    # Can use -g [group] before the flag to set group flags (ex. -g members pvp deny).
    # Can use PlaceholderAPI placeholders in string flags (ex. greeting, farewell).
    flags = [

        #IGNORED
        #   build
        #   blocked-cmds
        #   allowed-cmds
        #   exit-override
        #   farewell-title
        #   greeting-title
        #   weather-lock
        #   time-lock 
        #   healing, feeding, 'n shit

        "-g all pvp allow",
        "-g all tnt allow",
        "-g all potion-splash allow",
        "-g all firework-damage allow",
        "-g all exp-drops allow",
        "-g all creeper-explosion allow",
        "-g all enderdragon-block-damage allow",
        "-g all ghast-fireball allow",
        "-g all other-explosion allow",
        "-g all wither-damage allow",
        "-g all enderman-grief allow",
        "-g all snowman-trails allow",
        "-g all ravenger-grief allow",
        "-g all mob-spawning allow",
        "-g all pistons allow",
        "-g all fire-spread allow",
        "-g all lava-fire allow",
        "-g all lightning allow",
        "-g all snow-fall allow",
        "-g all snow-melt allow",
        "-g all ice-form allow",
        "-g all ice-melt allow",
        "-g all frosted-ice-melt allow",
        "-g all frosted-ice-form allow",
        "-g all mushroom-growth allow",
        "-g all leaf-decay allow",
        "-g all grass-growth allow",
        "-g all mycelium-spread allow",
        "-g all vine-growth allow",
        "-g all crop-growth allow",
        "-g all soil-dry allow",
        "-g all coral-fade allow",
        "-g all water-flow allow",
        "-g all lava-flow allow",
        "-g all send-chat allow",
        "-g all receive-chat allow",
        "-g all teleport-message Wzium...",
        "-g all invincible deny",
        "-g all fall-damage allow",
        "-g all natural-health-regen allow",
        "-g all natural-hunger-drain allow",
        "-g all entry allow",
        "-g all exit allow",
        "-g all exit-via-teleport allow",
        "-g all enderpearl allow",
        "-g all chorus-fruit-teleport allow",
        "-g all greeting &lWkraczasz na Kuboid gracza &b&l%player%",
        "-g all farewell &lOpuszczasz Kuboid gracza &b&l%player%",
        "-g all game-mode survival",
        "-g all deny-message &cNie twoje, nie dotykaj!",
        "-g all mob-damage allow",
        "-g all exit-deny-message jesli widzisz ta wiadomosc cos poszlo bardzo okropnie nie tak i nie mozesz opuscic kuboidu w ktorym sie znajdujesz NATYCHMIAST ZANOTUJ KOORDYNATY I OPUSC SERWER I POWIADOM ADMINISTRACJE if you can see this message someting went terribly wrong and you are stuck in the cuboid youre currently in REMEMBER THE CORDS AND LEAVE THE SERVER AND CONTACT THE ADMINS IMMIEDIATLEY",
        "-g all entry-deny-message jesli widzisz ta wiadomosc cos poszlo bardzo okropnie nie tak i nie mozesz wejsc na ten kuboid NATYCHMIAST ZANOTUJ KOORDYNATY I OPUSC SERWER I POWIADOM ADMINISTRACJE if you can see this message someting went terribly wrong and you are stuck in the cuboid youre currently in REMEMBER THE CORDS AND LEAVE THE SERVER AND CONTACT THE ADMINS IMMIEDIATLEY",

        "-g members block-break allow",
        "-g nonmembers block-break deny",
        "-g members block-place allow",
        "-g nonmembers block-place deny",
        "-g members use allow",
        "-g nonmembers use allow",
        "-g members interact allow",
        "-g nonmembers interact allow",
        "-g members damage-animals allow",
        "-g nonmembers damage-animals allow",
        "-g members sleep allow",
        "-g nonmembers sleep allow",
        "-g members respawn-anchors allow",
        "-g nonmembers respawn-anchors allow",
        "-g members chest-access allow",
        "-g nonmembers chest-access deny",
        "-g members vehicle-place allow",
        "-g nonmembers vehicle-place allow",
        "-g members vehicle-destroy allow",
        "-g nonmembers vehicle-destroy allow",
        "-g members lighter allow",
        "-g nonmembers lighter deny",
        "-g members ride allow",
        "-g nonmembers ride allow",
        "-g members item-frame-rotation allow",
        "-g nonmembers item-frame-rotation allow",
        "-g members block-trampling allow",
        "-g nonmembers block-trampling deny",
        "-g members use-anvil allow",
        "-g nonmembers use-anvil allow",
        "-g members item-pickup allow",
        "-g nonmembers item-pickup allow",
        "-g members item-drop allow",
        "-g nonmembers item-drop allow",    
        "-g members entity-painting-destroy allow",
        "-g nonmembers entity-painting-destroy deny",
        "-g members entity-item-frame-destroy deny",
        "-g nonmembers entity-item-frame-destroy deny",
        
        "-g members notify-enter False",
        "-g owners notify-enter False",
        "-g members notify-leave False",
        "-g owners notify-leave False"
    ]

    # List all the flags that can be set by region owners.
    # If you want to whitelist the groups that can be set as well, use -g (ex. "-g all,members pvp" restricts it to no group flag, and members group)
    # "-g all pvp" - Prevents players setting the group to nonmembers, and being invulnerable to attacks.
    #
    # -g cheatsheet:
    #   all                 - randoms, members and owners
    #   members             - members and owners
    #   owners              - owners
    #   nonmembers          - randoms
    #   nonowners           - members and randoms
    #   nonmembers,members  - what randos can do, and what's member-only

    allowed_flags = [
        #"-g nonmembers,members build",
        "-g nonmembers,members block-break",
        "-g nonmembers,members block-place",
        "-g nonmembers,members use",
        "-g nonmembers,members interact",
        "-g nonmembers,members damage-animals", 
        "-g all pvp",
        "-g nonmembers,members sleep",
        "-g nonmembers,members respawn-anchors",
        #"tnt",
        "-g nonmembers,members chest-access",
        "-g nonmembers,members vehicle-place",
        "-g nonmembers,members vehicle-destroy",
        "-g nonmembers,members lighter",
        "-g nonmembers,members ride",
        "-g all potion-splash",
        "-g nonmembers,members item-frame-rotation",
        "-g nonmembers,members block-trampling",
        "-g all firework-damage",
        "-g nonmembers,members use-anvil",
        "-g nonmembers,members item-pickup",
        "-g nonmembers,members item-drop",
        "-g all exp-drops",
        "-g nonmembers,members mob-damage",
        "-g all creeper-explosion",
        "-g all enderdragon-block-damage",
        "-g all ghast-fireball",
        #"-g all other-explosion",
        "-g all wither-damage",
        "-g all enderman-grief",
        "-g all snowman-trails",
        "-g all ravenger-grief",
        "-g nonmembers,members entity-painting-destroy",
        "-g nonmembers,members entity-item-frame-destroy",
        "-g all mob-spawning",
        "-g all pistons",
        "-g all fire-spread",
        "-g all lava-fire",
        "-g all lightning",
        "-g all snow-fall",
        "-g all snow-melt",
        "-g all ice-form",
        "-g all ice-melt",
        "-g all frosted-ice-melt",
        "-g all frosted-ice-form",
        "-g all mushroom-growth",
        "-g all leaf-decay",
        "-g all grass-growth",
        "-g all mycelium-spread",
        "-g all vine-growth",
        "-g all crop-growth",
        "-g all soil-dry",
        "-g all coral-fade",
        "-g all water-flow",
        "-g all lava-flow",
        #"weather-lock",
        #"time-lock",
        #"send-chat",
        #"receive-chat",
        #"blocked-cmds",
        #"allowed-cmds",
        #"-g all teleport",
        #"spawn",
        "-g all teleport-message",
        #"invincible",
        #"fall-damage",
        #"natural-health-regen",
        #"natural-hunger-drain",
        #"entry",
        #"exit",
        #"exit-override",
        #"exit-via-teleport",
        "-g all enderpearl",
        "-g all chorus-fruit-teleport",
        "-g all greeting",
        "-g all farewell",
        "-g all greeting-title",
        "-g all farewell-title",
        #"-g owners,members notify-enter",
        #"-g owners,members notify-leave",
        #"game-mode",
        #"heal-delay",
        #"heal-amount",
        #"heal-min-health",
        #"heal-max-health",
        #"feed-delay",
        #"feed-amount",
        #"feed-min-hunger",
        #"feed-max-hunger",
        "-g all deny-message",
        #"exit-deny-message",
        #"entry-deny-message"
    ]

    # Which flags to hide from /ps info
    hidden_flags_from_info = [
        "ps-merged-regions",
        "ps-merged-regions-types",
        "ps-price",
        "ps-tenant",
        "ps-rent-period",
        "ps-rent-last-paid",
        "ps-for-sale",
        "ps-rent-settings",
        "ps-tax-payments-due",
        "ps-tax-last-payment-added",
        "ps-tax-autopayer"
    ]
```

**Comment by GuzioMG** [comment]
Thanks for the help in advance!

**Comment by espidev** [comment]
WorldGuard only accepts one value per flag, so this for example:

```
"-g members entity-item-frame-destroy deny",
"-g nonmembers entity-item-frame-destroy deny",
```

would only take on the second line. You would probably want to instead do `-g nonowners entity-item-frame-destroy deny` in this case.

The errors are a separate issue, they are caused by `-g all`, which doesn't seem to exist in WorldGuard's flag parsing mechanism I'm using... Try removing all of the `-g all` prefixes you have on the flags, it's the same intended effect without them.

**Comment by GuzioMG** [comment]
> WorldGuard only accepts one value per flag, so this for example: `"-g members entity-item-frame-destroy deny",
"-g nonmembers entity-item-frame-destroy deny",` it would only take on the second line. You would probably want to instead do -g nonowners entity-item-frame-destroy deny in this case.
 
Oh, thanks for pointing that out. I actually haven't realized that I had `deny` set for both members and nonmembers in `entity-item-frame-destroy`. Fixed that, now I have `allow` for members and `deny` for nonmembers. As for other ones where I had `allow` set for both members and nonmembers... I just merged them into one, with no group parameters and set that one to `allow`. 

> The errors are a separate issue, they are caused by -g all, which doesn't seem to exist in WorldGuard's flag parsing mechanism I'm using... Try removing all of the -g all prefixes you have on the flags, it's the same intended effect without them.

Thank you, the errors are gone! :)
